airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -453,10 +456,12 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
               MssqlCdcTargetPosition.getTargetPosition(database, sourceConfig.get(JdbcUtils.DATABASE_KEY).asText()), true, firstRecordWaitTime,
               OptionalInt.empty());
 
+      final MssqlCdcConnectorMetadataInjector mssqlCdcConnectorMetadataInjector = MssqlCdcConnectorMetadataInjector.getInstance(emittedAt);
```

Comment:
One quick thought I had was : 
if this is now stateful, do we want to declare this in the constructor to prevent other parts of the code from declaring multiple versions of the CdcConnectorMetadataInjector?

Reply:
@akashkulk  I actually already turned it into a singleton. Refer to `MssqlCdcConnectorMetadataInjector.java`